### PR TITLE
adding cugraph-ops

### PR DESCRIPTION
**Pull request by BradReesWork**
The goal is to have a unified docs area for all graph work.  This PR is the first step in getting cugraph-ops included in the main graph doc area

### CODE REVIEW

File: dependencies.yaml [hunk 1]
```diff
@@ -251,6 +251,7 @@ dependencies:
           - sphinx-markdown-tables
           - sphinx<6
           - sphinxcontrib-websupport
+          - pylibcugraphops=23.6.*
```
Comment (ajschmidt8) [inline dependencies.yaml:254]: can you add an entry for this file in `update-version.sh`?

Comment (BradReesWork) [reply dependencies.yaml:254]: update and in latest push

**Review by ajschmidt8 — APPROVED** [review]

File: docs/cugraph/source/api_docs/cugraph-ops/index.rst [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+cugraph-ops API reference
+=========================
+
+This page provides a list of all publicly accessible modules, methods and classes through `pylibcugraphops.*` namespace.
+
+.. toctree::
+    :maxdepth: 2
+    :caption: API Documentation
+
+    graph_types
```
Comment (MatthiasKohl) [inline docs/cugraph/source/api_docs/cugraph-ops/index.rst:11]: should also add pytorch here at the top IMO (just noticed that our own docs have them at the end, but pytorch is the main API for end-users so this should be the first thing to see)

```suggestion
    pytorch
    graph_types
```

**Review by MatthiasKohl — CHANGES_REQUESTED** [review]
Like @tingyu66 pointed out, the pytorch file naming is incorrect.

Additionally, I'm not sure what the plan is, but the C++ API docs are not found here: is the plan to just not add them at all, add them in a later PR, or should they be added immediately here?

File: docs/cugraph/source/api_docs/cugraph-ops/index.rst (hunk 2 above)
Comment (BradReesWork) [reply docs/cugraph/source/api_docs/cugraph-ops/index.rst:11]: Adding the C++ API is an upcoming step.   Working on covering from Doxogen to Sphinx.  Once I have the process setup for the other cugraph C code, then I'll be rolling in cugraph-ops.

**Review by tingyu66 — APPROVED** [review]
👍

File: docs/cugraph/source/api_docs/cugraph-ops/index.rst (hunk 2 above)
Comment (MatthiasKohl) [reply docs/cugraph/source/api_docs/cugraph-ops/index.rst:11]: I believe you now added the pytorch entry twice, as well as not adding it at the top as suggested above.
Let's just remove one of these instances, or completely remove this if it doesn't pass CI

File: docs/cugraph/source/index.rst [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
-NetworkX Graphs, or even CuPy or SciPy sparse Matrices.
+Welcome to RAPIDS Graph documentation
+=====================================
+RAPIDS Graph covers a range of libraries and packages, that includes:
+
+__cugraph__: focuses on GPU accelerated graph algorithms.
+
+__cugraph-ops__: focuses on GNN aggregators and operators.
+
+__cugraph-pg__: focuses on providing support for Property Graph (and future Knowledge Graph).
```
Comment (rlratzel) [inline docs/cugraph/source/index.rst:9]: I don't think this exists yet, is that okay?

**Review by rlratzel — COMMENTED** [review]
Overall looks good to me. I do think it's a little odd to add docs to components that live in another repo here, but I can't think of a better solution now.

File: docs/cugraph/source/index.rst (hunk 3 above)
Comment (BradReesWork) [reply docs/cugraph/source/index.rst:9]: yep.   That will be discussed in the section on PG and I will be updating this intro section

**Comment by BradReesWork** [comment]
> a little odd to add docs to components that live in another repo

I have been pondering how to create a unified docs page that brings in all the various sections.  Most examples from other products use a central repo that clones all the other repos into one so that docs can be created.  I'm trying to avoid that.

Yes it is odd to have stuff from another repo - but it all falls under Graph

**Review by acostadon — APPROVED** [review]
Looks good to me

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**